stdlib/public/core/StringStorageBridge.swift

```diff
@@ -17,6 +17,28 @@ import SwiftShims
 internal let _cocoaASCIIEncoding:UInt = 1 /* NSASCIIStringEncoding */
 internal let _cocoaUTF8Encoding:UInt = 4 /* NSUTF8StringEncoding */
 
+extension String {
+  @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+  @_spi(Foundation)
+  public init?(_nativeStorage: AnyObject) {
```

Comment:
What is this used for, and is this tested anywhere? Comments? That's a lot of unsafe unchecked downcasts and it's not clear to me what is assumed about `_nativeStorage`.